src/melee.cpp

```diff
@@ -1402,24 +1462,35 @@ bool player::block_hit(Creature *source, body_part &bp_hit, damage_instance &dam
     // but it still counts as a block even if it absorbs all the damage.
     float total_phys_block = mabuff_block_bonus();
     bool conductive_weapon = weapon.conductive();
+    bool conductive_armor = armor.conductive();
```

Comment:
You don't need to separate armor and weapon here. Have some function above select the better of the two. You can name the variable `shield` - it would be better than `armor` because armor is more passive.